### PR TITLE
Unit localization

### PR DESCRIPTION
**Pull request by samuelgarcia**
1. Use existing sparsity for unit location
 2. Add location with max channel

### CODE REVIEW

**Review by zm711 — COMMENTED** [review]
Thanks for this. We were actually having some weird issues and wanted to check the max channel only so this is perfect for us.

**Review by yger — APPROVED** [review]
This is good to me, good catch. Indeed, this make sense to use the internal, and possible more fancy sparsity of the templates for unit locations

File: src/spikeinterface/postprocessing/localization_tools.py [hunk 1]
```diff
@@ -667,14 +667,14 @@ def compute_location_max_channel(
     """
     Localize a unit using max channel.
 
-    This use inetrnally get_template_extremum_channel()
+    This uses interrnally `get_template_extremum_channel()`
```
Comment (alejoe91) [inline src/spikeinterface/postprocessing/localization_tools.py:670]: Nice to correct a typo with another smaller typo! You almost got us all fooled :P
```suggestion
    This uses internally `get_template_extremum_channel()`
```

File: src/spikeinterface/postprocessing/localization_tools.py [hunk 2]
```diff
@@ -650,8 +658,55 @@ def get_convolution_weights(
...
+        A list of unit_id to restrict the computation
+    peak_sign :  "neg" | "pos" | "both"
+        Sign of the template to find extremum channels
+    mode : "extremum" | "at_index" | "peak_to_peak", default: "at_index"
+        Where the amplitude is computed
+        * "extremum" : take the peak value (max or min depending on `peak_sign`)
+        * "at_index" : take value at `nbefore` index
+        * "peak_to_peak" : take the peak-to-peak amplitude
+
+            Returns
```
Comment (alejoe91) [inline src/spikeinterface/postprocessing/localization_tools.py:687]:
```suggestion
    Returns
```

File: src/spikeinterface/postprocessing/localization_tools.py [hunk 3]
```diff
@@ -650,8 +658,55 @@ def get_convolution_weights(
...
+    -------
+    unit_location: np.ndarray
```
Comment (alejoe91) [inline src/spikeinterface/postprocessing/localization_tools.py:689]:
```suggestion
    unit_locations: np.ndarray
```

File: src/spikeinterface/postprocessing/localization_tools.py [hunk 4]
```diff
@@ -650,8 +658,55 @@ def get_convolution_weights(
...
+    contact_locations = templates_or_sorting_analyzer.get_channel_locations()
+    if unit_ids is None:
+        unit_ids = templates_or_sorting_analyzer.unit_ids
+    else:
+        unit_ids = np.asarray(unit_ids)
+    unit_location = np.zeros((unit_ids.size, 2), dtype="float32")
+    for i, unit_id in enumerate(unit_ids):
+        unit_location[i, :] = contact_locations[extremum_channels_index[unit_id]]
+
+    return unit_location
```
Comment (alejoe91) [inline src/spikeinterface/postprocessing/localization_tools.py:704]:
```suggestion
    unit_locations = np.zeros((unit_ids.size, 2), dtype="float32")
    for i, unit_id in enumerate(unit_ids):
        unit_locations[i, :] = contact_locations[extremum_channels_index[unit_id]]

    return unit_locations
```

File: src/spikeinterface/postprocessing/localization_tools.py (hunk 1 above)
Comment (zm711) [reply src/spikeinterface/postprocessing/localization_tools.py:670]: That was my fault. Sometimes I type too fast too! my bad :)

**Review by alejoe91 — APPROVED** [review]

**Pull request merged**